dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/merge/orgunit/DefaultOrgUnitMergeService.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2004-2021, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.merge.orgunit;
+
+import java.util.Iterator;
+import java.util.Set;
+
+import javax.transaction.Transactional;
+
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.merge.orgunit.handler.AnalyticalObjectOrgUnitMergeHandler;
+import org.hisp.dhis.merge.orgunit.handler.DataOrgUnitMergeHandler;
+import org.hisp.dhis.merge.orgunit.handler.MetadataOrgUnitMergeHandler;
+import org.hisp.dhis.merge.orgunit.handler.TrackerOrgUnitMergeHandler;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+
+/**
+ * Main class for org unit merge.
+ *
+ * @author Lars Helge Overland
+ */
+@Service
+public class DefaultOrgUnitMergeService
+    implements OrgUnitMergeService
+{
+    private final MetadataOrgUnitMergeHandler metadataHandler;
+
+    private final AnalyticalObjectOrgUnitMergeHandler analyticalObjectHandler;
+
+    private final DataOrgUnitMergeHandler dataHandler;
+
+    private final TrackerOrgUnitMergeHandler trackerHandler;
```

Comment:
Feel like these should not be fields as `handlers` is the way to access them once they have been build into the list. Couldn't those references be passed to the `getMergeHandlers()` method to avoid the fields?

Reply:
Hm I think the reason is that they have to be injected as dependencies by Spring?

Reply:
Spring is happy with them occurring as constructor arguments. No fields needed.

Reply:
I see what you mean now.